.github/workflows/snap.yaml

```diff
@@ -17,7 +17,7 @@ on:
       branch-name:
         required: false
         type: string
-        default: "main"
+        default: '${{ github.ref }}'
```

Comment:
It doesn't solve the issue that the publish job is going to be triggered in every situation.

Reply:
No indeed it doesn't. This should be tackled in a subsequent pr.